src/playercards/spawnbag/SpawnBag.ttslua

```diff
@@ -0,0 +1,202 @@
+      if (isInRecallZone(obj)) then
+        trash.putObject(obj)
+      end
+      placedObjectGuids[guid] = nil
+    end
+  end
+
+  trash.destruct()
+  -- We've recalled everything we can, some cards may have been moved out of the
+  -- card area.  Just reset at this point.
+  placedSpecs = { }
+  placedObjectGuids = { }
+  recallZone = nil
+end
+
+-- Callback for when an object has been spawned.  Tracks the object for later recall and updates the
+-- recall zone.
+function recordPlacedObject(spawned)
+  placedObjectGuids[spawned.getGUID()] = true
+  expandRecallZone(spawned)
+end
+
+-- Expands the current recall zone based on the position of the given object.  The recall zone will
+-- be maintained as the bounding box of the extreme object positions, plus a small amount of buffer
+function expandRecallZone(spawnedCard)
+  local pos = spawnedCard.getPosition()
+  if (recallZone == nil) then
+    -- First card out of the bag, initialize surrounding that
+    recallZone = { }
+    recallZone.upperLeft = { x = pos.x + RECALL_BUFFER_X, z = pos.z + RECALL_BUFFER_Z }
+    recallZone.lowerRight = { x = pos.x - RECALL_BUFFER_X, z = pos.z - RECALL_BUFFER_Z }
+    return
+  else
+    if (pos.x > recallZone.upperLeft.x) then
+      recallZone.upperLeft.x = pos.x + RECALL_BUFFER_X
+    end
+    if (pos.x < recallZone.lowerRight.x) then
+      recallZone.upperLeft.x = pos.x - RECALL_BUFFER_X
+    end
+    if (pos.z > recallZone.upperLeft.z) then
+      recallZone.upperLeft.z = pos.z + RECALL_BUFFER_Z
+    end
+    if (pos.z < recallZone.lowerRight.z) then
+      recallZone.lowerRight.z = pos.z - RECALL_BUFFER_Z
+    end
+  end
+  if (SHOW_RECALL_ZONE) then
+    local y = 1.5
+    local thick = 0.05
+    Global.setVectorLines({
+      {
+          points    = { {recallZone.upperLeft.x,y,recallZone.upperLeft.z}, {recallZone.upperLeft.x,y,recallZone.lowerRight.z} },
+          color     = {1,0,0},
+          thickness = thick,
+          rotation  = {0,0,0},
+      },
+      {
+          points    = { {recallZone.upperLeft.x,y,recallZone.lowerRight.z}, {recallZone.lowerRight.x,y,recallZone.lowerRight.z} },
+          color     = {1,0,0},
+          thickness = thick,
+          rotation  = {0,0,0},
+      },
+      {
+          points    = { {recallZone.lowerRight.x,y,recallZone.lowerRight.z}, {recallZone.lowerRight.x,y,recallZone.upperLeft.z} },
+          color     = {1,0,0},
+          thickness = thick,
+          rotation  = {0,0,0},
+      },
+      {
+          points    = { {recallZone.lowerRight.x,y,recallZone.upperLeft.z}, {recallZone.upperLeft.x,y,recallZone.upperLeft.z} },
+          color     = {1,0,0},
+          thickness = thick,
+          rotation  = {0,0,0},
+      },
+      })
+  end
+end
+
+-- Checks to see if the given object is in the current recall zone.  If there isn't a recall zone,
+-- will return true so that everything can be easily cleaned up.
+function isInRecallZone(obj)
+  if (recallZone == nil) then
+    return true
```

Comment:
Why would you want to return true if there is no zone defined?

Reply:
This is basically an error condition, and fails to "recall everything" rather than "recall nothing".  IMHO is a better default behavior to be overly aggressive than just do nothing.